cli/ssh.go

```diff
@@ -335,10 +329,87 @@ func (r *RootCmd) ssh() *clibase.Cmd {
 			Description: "Specifies whether to wait for a workspace to become ready before logging in (only applicable when the login before ready option has not been enabled). Note that the workspace agent may still be in the process of executing the startup script and the workspace may be in an incomplete state.",
 			Value:       clibase.BoolOf(&noWait),
 		},
+		{
+			Flag:        "log-dir",
+			Default:     os.TempDir(),
+			Description: "Specify the location for the log files.",
+			Env:         "CODER_SSH_LOG_DIR",
+			Value:       clibase.StringOf(&logDir),
+		},
+		{
+			Flag:          "log-to-file",
+			FlagShorthand: "l",
+			Env:           "CODER_SSH_LOG_TO_FILE",
+			Description:   "Enable diagnostic logging to file.",
+			Value:         clibase.BoolOf(&logToFile),
+		},
 	}
 	return cmd
 }
 
+// watchAndClose Enures closer is called if the context is canceled or
+// the workspace reaches the stopped state.
+//
+// Watching the stopped state is a work-around for cases
+// where the agent is not gracefully shut down and the
+// connection is left open. If, for instance, the networking
+// is stopped before the agent is shut down, the disconnect
+// will usually not propagate.
+//
+// See: https://github.com/coder/coder/issues/6180
+func watchAndClose(ctx context.Context, closer func() error, logger slog.Logger, client *codersdk.Client, workspace codersdk.Workspace) {
+	// Ensure session is ended on both context cancellation
+	// and workspace stop.
+	defer func() {
+		_ = closer()
```

Comment:
nit: as you're logging all return errors, this one is swallowed. Maybe leave a comment if it is not relevant?